packages/wonder-blocks-tokens/src/tokens/semantic-color.ts

```diff
@@ -1,18 +1,87 @@
 import {color} from "./color";
 
+const border = {
+    primary: color.fadedOffBlack16,
+    subtle: color.fadedOffBlack8,
+    strong: color.fadedOffBlack50,
+    inverse: color.white,
+};
+
 export const semanticColor = {
     /**
      * For buttons, links, and controls to communicate the presence and meaning
      * of interaction.
      */
     action: {
-        primary: {
-            default: color.blue,
-            active: color.activeBlue,
+        // For primary actions
+        progressive: {
+            default: {
+                border: "transparent",
+                background: color.blue,
+                foreground: color.white,
+            },
+            hover: {
+                border: color.blue,
+                background: color.blue,
+                foreground: color.white,
+            },
+            press: {
+                border: color.activeBlue,
+                background: color.activeBlue,
+                foreground: color.white,
+            },
+        },
+        // Inverse is meant for use on secondary controls.
+        progressiveInverse: {
+            default: {
+                border: border.strong,
+                background: color.white,
+                foreground: color.blue,
+            },
+            hover: {
+                border: color.blue,
+                background: color.white,
+                foreground: color.blue,
+            },
+            press: {
+                border: color.activeBlue,
+                background: color.fadedBlue,
+                foreground: color.activeBlue,
+            },
         },
         destructive: {
-            default: color.red,
-            active: color.activeRed,
+            default: {
+                border: "transparent",
+                background: color.red,
+                foreground: color.white,
+            },
+            hover: {
+                border: color.red,
+                background: color.white,
+                foreground: color.red,
+            },
+            press: {
+                border: color.activeRed,
+                background: color.activeRed,
+                foreground: color.white,
+            },
+        },
+        destructiveInverse: {
+            default: {
+                border: border.strong,
+                background: color.white,
+                foreground: color.red,
+            },
+            hover: {
+                border: color.red,
+                background: color.white,
+                foreground: color.red,
+            },
+            press: {
+                border: color.activeRed,
+                background: color.fadedRed,
+                foreground: color.activeRed,
+            },
         },
         disabled: {
```

Comment:
Should the tokens associated with the `disabled` state also have the backgorund, foreground, border structure?

Reply:
It might be in the future, but for now I'm leaving them out of every specific semantic global token and I'm defining them inside `Button` for now. Buttons need simplification and before doing that, I prefer to keep it this way. FWIW Design also has this structure, but this could change in the next weeks.